### PR TITLE
[QA-209] Add scripts for BTM and TxMA

### PR DESCRIPTION
**Pull request by paulfloodgds**
### What?
Added TXMA and BTM Send Events Scripts - Run successfully locally and on local version of the devplatform-performance

#### Changes:
- First commit of TXMA and BTM test.ts

---

### Why?
Verify both scripts in the Perf Platform

---

### CODE REVIEW

File: deploy/scripts/src/btm/test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { type Options } from 'k6/options'
+
+import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
+
+import { uuidv4 } from 'https://jslib.k6.io/k6-utils/1.4.0/index.js'
+import { AWSConfig, SQSClient } from 'https://jslib.k6.io/aws/0.7.2/sqs.js'
```
Comment (Tom-Dann) [inline deploy/scripts/src/btm/test.ts:6]: These will need to be added as files to the repository as we cannot run external code from the performance test stack.

I'd suggest adding the files to a new folder under `/deploy/scripts/common/utils/jslib/`, add any copyright licensing as a comment at the top of the script and then change the imports here to use relative paths.

File: deploy/scripts/src/txma/test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { type Options } from 'k6/options'
+
+import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
+
+import { uuidv4 } from 'https://jslib.k6.io/k6-utils/1.4.0/index.js'
+import { AWSConfig, SQSClient } from 'https://jslib.k6.io/aws/0.7.2/sqs.js'
```
Comment (Tom-Dann) [inline deploy/scripts/src/txma/test.ts:6]: The same comment about external file imports applies here

File: deploy/scripts/src/btm/test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    http_req_failed: ['rate<0.05'] // Error rate <5%
+  }
+}
+
+export function setup (): void {
+  describeProfile(loadProfile)
+}
+
+const env = {
+  sqs_queue: __ENV.BTM_AWS_SQS_QUEUE
```
Comment (Tom-Dann) [inline deploy/scripts/src/btm/test.ts:39]: Add environment variables to the BuildSpec in `deploy/template.yaml` script under the parameter store section

File: deploy/scripts/src/btm/test.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const awsConfig = new AWSConfig({
+  region: __ENV.BTM_AWS_REGION,
+  accessKeyId: __ENV.BTM_AWS_ACCESS_KEY_ID,
+  secretAccessKey: __ENV.BTM_AWS_SECRET_ACCESS_KEY,
+  sessionToken: __ENV.BTM_AWS_SESSION_TOKEN
```
Comment (Tom-Dann) [inline deploy/scripts/src/btm/test.ts:46]: Is the plan here to use temporary credentials to test?

I think longer term we need to look at a better way of delegating access or assume roles for accessing SQS queues

File: deploy/scripts/src/txma/test.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+const awsConfig = new AWSConfig({
+  region: __ENV.TXMA_AWS_REGION,
+  accessKeyId: __ENV.TXMA_AWS_ACCESS_KEY_ID,
+  secretAccessKey: __ENV.TXMA_AWS_SECRET_ACCESS_KEY,
+  sessionToken: __ENV.TXMA_AWS_SESSION_TOKEN
+})
+
+const eventData = {
+  payload: __ENV.TXMA_DATA_001
+  // payload: __ENV.BTM_DATA_001
```
Comment (Tom-Dann) [inline deploy/scripts/src/txma/test.ts:51]: Remove unused comments
```suggestion
```

File: deploy/scripts/src/btm/test.ts (hunk 4 above)
Comment (paulfloodgds) [reply deploy/scripts/src/btm/test.ts:46]: As discussed, yes to support the K6 AWS Lib plan to use temp credentials and look at a longer term solution to automate this step

File: deploy/scripts/src/txma/test.ts (hunk 5 above)
Comment (paulfloodgds) [reply deploy/scripts/src/txma/test.ts:51]: Removed

File: deploy/scripts/src/btm/test.ts (hunk 1 above)
Comment (paulfloodgds) [reply deploy/scripts/src/btm/test.ts]: kept the mobile folders structure of k6 to match existing Mobile folder structure

File: deploy/scripts/src/txma/test.ts (hunk 2 above)
Comment (paulfloodgds) [reply deploy/scripts/src/txma/test.ts]: Done

File: deploy/scripts/src/btm/test.ts (hunk 3 above)
Comment (paulfloodgds) [reply deploy/scripts/src/btm/test.ts:40]: Done

File: deploy/scripts/src/txma/test.ts (hunk 5 above)
Comment (paulfloodgds) [reply deploy/scripts/src/txma/test.ts]: Done

File: deploy/template.yaml [hunk 6]
```diff
@@ -297,6 +297,8 @@ Resources:
               MOBILE_BACKEND_URL: "/perfTest/mobile/backendUrl"
               MOBILE_FRONTEND_URL: "/perfTest/mobile/frontendUrl"
               MOBILE_TEST_CLIENT_EXECUTE_URL: "/perfTest/mobile/testClientUrl"
+              BTM_AWS_SQS_QUEUE: "/perfTest/txmxconsumeroutputqueuebilling"
+              TXMA_AWS_SQS_QUEUE: "/perfTest/txmxscaledauditeventsqueue"
```
Comment (Tom-Dann) [inline deploy/template.yaml:301]:
```suggestion
              DATA_BTM_SQS: "/perfTest/data/btm/consumerOutputSQS"
              DATA_TXMA_SQS: "/perfTest/data/txma/auditSQS"
```
Please update the names of these so we are following the format `<POD>_<TEAM>_<VAR_NAME>` or `<POD>_<VAR_NAME>` for the environment variables; and `/perfTest/<pod>/<team>/<varName>` or `/perfTest/<pod>/<varName>` for the parameter store paths.

And then move them so the list remains ordered alphabetically

File: deploy/scripts/src/btm/test.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import { type Options } from 'k6/options'
+
+import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
+
+import { uuidv4 } from '../common/utils/k6/k6-utils'
+
+import { AWSConfig, SQSClient } from '../common/utils/aws/sqs'
```
Comment (Tom-Dann) [inline deploy/scripts/src/btm/test.ts:7]:
```suggestion
import { type Options } from 'k6/options'
import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
import { uuidv4 } from '../common/utils/k6/k6-utils'
import { AWSConfig, SQSClient } from '../common/utils/aws/sqs'
```
Minor whitespace tidy up on the imports

File: deploy/scripts/src/txma/test.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
+import { type Options } from 'k6/options'
+
+import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
+
+import { uuidv4 } from '../common/utils/k6/k6-utils'
+
+import { AWSConfig, SQSClient } from '../common/utils/aws/sqs'
+
```
Comment (Tom-Dann) [inline deploy/scripts/src/txma/test.ts:8]:
```suggestion
import { type Options } from 'k6/options'
import { selectProfile, type ProfileList, describeProfile } from '../common/utils/config/load-profiles'
import { uuidv4 } from '../common/utils/k6/k6-utils'
import { AWSConfig, SQSClient } from '../common/utils/aws/sqs'
```
Minor whitespace tidy up on the imports

Comment (paulfloodgds) [reply deploy/scripts/src/txma/test.ts:8]: Done

File: deploy/scripts/src/btm/test.ts (hunk 7 above)
Comment (paulfloodgds) [reply deploy/scripts/src/btm/test.ts:7]: Done

File: deploy/template.yaml (hunk 6 above)
Comment (paulfloodgds) [reply deploy/template.yaml:301]: Done

**Pull request merged**